src/pulsar.appup.src

```diff
@@ -1,139 +1,360 @@
-      {load_module,pulsar_consumers,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_producers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_client,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumer,brutal_purge,soft_purge,[]},
-      {update, pulsar_producer, {advanced, []}},
-      {load_module,pulsar_protocol_frame,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_socket,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_utils,brutal_purge,soft_purge,[]}]},
-    {"0.5.4",
-     [{load_module,pulsar,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_producers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_socket,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_protocol_frame,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_producers,brutal_purge,soft_purge,[]},
-      {update, pulsar_producer, {advanced, []}},
-      {load_module,pulsar_consumers,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumer,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_client,brutal_purge,soft_purge,[]},
-      {delete_module,pulsar_utils}]},
-    {<<"0.5.[0-3]">>,
-     [{load_module,pulsar,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_producers_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_protocol_frame,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_producers,brutal_purge,soft_purge,[]},
-      {update, pulsar_producer, {advanced, []}},
-      {load_module,pulsar_consumers,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_consumer,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_client,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_client_sup,brutal_purge,soft_purge,[]},
-      {load_module,pulsar_api,brutal_purge,soft_purge,[]},
-      {delete_module,pulsar_socket},
-      {delete_module,pulsar_utils}]}]
-}.
+ [{"0.6.4",
+   [{add_module,pulsar_relup},
+    {apply,{pulsar_relup,suspend_producers,[]}},
+    {load_module,pulsar_producer,brutal_purge,soft_purge,[]},
+    {apply,{pulsar_relup,change_producers_up,[{0,7,0},{0,6,4}]}},
+    {apply,{pulsar_relup,resume_producers,[]}},
+    {load_module,pulsar_utils,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_sup,brutal_purge,soft_purge,[]},
+    {load_module,pulsar,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_socket,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_consumer,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_producers_sup,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_consumers_sup,brutal_purge,soft_purge,[]},
+    {load_module,pulsar_client_sup,brutal_purge,soft_purge,[]}]},
```

Comment:
load other modules before resuming producers.

Reply:
:heavy_check_mark: 

Moved the `pulsar_relup` operations block to the end of the instructions list.